# HG changeset patch
# User Shai Ayal <shaiay@users.sourceforge.net>
# Date 1205772696 -7200
#      Mon Mar 17 18:51:36 2008 +0200
# Node ID 7397a0026ca855f8d38574eeda58a8d2166494b8
# Parent  67edbcb19665446df26053e3670a6f31078c0897
make the fltk_backend canvas the size of figure.position

diff --git a/src/graphics/ChangeLog b/src/graphics/ChangeLog
--- a/src/graphics/ChangeLog
+++ b/src/graphics/ChangeLog
@@ -1,8 +1,13 @@
+2008-03-17  Shai Ayal  <shaiay@users.sourceforge.net>
+
+	* fltk_backend/fltk_backend.cc (plot_window::resize,
+	plot_window::draw): make canvas the size of figure.position
+
 2008-03-09  Shai Ayal  <shaiay@users.sourceforge.net>
 
 	* fltk_backend/fltk_backend.cc (plot_window::handle): add zoom
 	stack
 
 2008-03-06  Shai Ayal  <shaiay@users.sourceforge.net>
 
 	* fltk_backend/fltk_backend.cc (plot_window::handle): Add handling
diff --git a/src/graphics/fltk_backend/fltk_backend.cc b/src/graphics/fltk_backend/fltk_backend.cc
--- a/src/graphics/fltk_backend/fltk_backend.cc
+++ b/src/graphics/fltk_backend/fltk_backend.cc
@@ -342,25 +342,25 @@ private:
   void resize (int _x,int _y,int _w,int _h) 
   {
     Fl_Window::resize (_x, _y, _w, _h);
 
     Matrix pos (1,4,0);
     pos(0) = _x;
     pos(1) = _y;
     pos(2) = _w;
-    pos(3) = _h;
+    pos(3) = _h - status_h;
 
     fp.set_position (pos);
   }
 
   void draw (void)
   {
     Matrix pos = fp.get_position ().matrix_value ();
-    Fl_Window::resize (pos(0), pos(1) , pos(2), pos(3));
+    Fl_Window::resize (pos(0), pos(1) , pos(2), pos(3) + status_h);
 
     return Fl_Window::draw ();
   }
  
   int handle (int event) {
     static int px0,py0;
     static graphics_handle h0 = graphics_handle ();
 
